lib/crawly/worker.ex

```diff
@@ -76,16 +76,26 @@ defmodule Crawly.Worker do
              spider_name: atom(),
              response: HTTPoison.Response.t(),
              result: {:ok, {response, spider_name}} | {:error, term()}
-  defp get_response({request, spider_name}) do
+  def get_response({request, spider_name}) do
     # check if spider-level fetcher is set. Overrides the globally configured fetcher.
     # if not set, log warning for explicit config preferred,
     # get the globally-configured fetcher. Defaults to HTTPoisonFetcher
+
     {fetcher, options} =
-      Crawly.Utils.get_settings(
-        :fetcher,
-        spider_name,
-        {Crawly.Fetchers.HTTPoisonFetcher, []}
-      )
+      case Crawly.Utils.get_settings(
+             :fetcher,
+             spider_name,
+             {Crawly.Fetchers.HTTPoisonFetcher, []}
+           ) do
+        {module, args} ->
```

Comment:
can you add in an `is_keyword` guard to ensure that args is a keyword list.

Reply:
There's no `is_keyword` guard, checked around, could you send a pointer?

Reply:
Ah my bad, you can use is_list instead

Reply:
Done